client/lib/plans/index.js

```diff
@@ -60,10 +61,10 @@ export function canUpgradeToPlan( planKey, site ) {
 	return get( getPlan( planKey ), 'availableFor', () => false )( plan );
 }
 
-export function getUpgradePlanSlugFromPath( path, site ) {
-	return find( Object.keys( PLANS_LIST ), planKey => (
-		( planKey === path || getPlanPath( planKey ) === path ) &&
-		canUpgradeToPlan( planKey, site )
+export function getUpgradePlanSlugFromPath( path, isJetpack ) {
```

Comment:
I don't think we do. A function that maps a slug to a plan has no business checking access. That's a job for the server.